src/test/java/BQJDBC/QueryResultTest/QueryResultTest.java

```diff
@@ -110,6 +111,12 @@ public void NewConnection() {
         NewConnection(null);
     }
 
+    @After
+    public void TeardownConnection() throws SQLException {
+        QueryResultTest.con.close();
+        QueryResultTest.con = null;
+    }
+
```

Comment:
Depending on the order, the connection was being reused and it may have been created in legacy or non-legacy mode.
Certain tests used the legacy way of referring to projects
```sql
SELECT TOP(word, 10), COUNT(*) FROM publicdata:samples.shakespeare
```